xcdat/dataset.py

```diff
@@ -359,6 +367,95 @@ def decode_time(dataset: xr.Dataset) -> xr.Dataset:
     return ds
 
 
+def _is_paths_to_xml(paths: Union[str, pathlib.Path]) -> bool:
+    """Checks if the ``paths`` argument is a path to an XML file.
+
+    Parameters
+    ----------
+    paths : Union[str, pathlib.Path]
+        A string or pathlib.Path represnting a file path.
+
+    Returns
+    -------
+    bool
+    """
+    if isinstance(paths, str):
+        return paths.split(".")[-1] == "xml"
+    elif isinstance(paths, pathlib.Path):
+        return paths.parts[-1].endswith("xml")
+
+
+def _parse_xml_for_nc_glob(xml_path: str) -> str:
+    """Parses a CDAT XML file for a glob of `*.nc` paths.
+
+    The CDAT "Climate Data Markup Language" (CDML) is a dialect of XML with a
+    defined set of attributes. The "directory" attribute pointing to a directory
+    of `.nc` files is extracted from the specified XML file. Refer to [6]_ for
+    more information on CDML.
+
+    Parameters
+    ----------
+    xml_path : str
+        The CDML XML file path.
+
+    Returns
+    -------
+    str
+        A glob of `*.nc` paths in the directory.
+
+    Notes
+    -----
+    CDML is a deprecated XML dialect that is still used by current and former
+    users in the CDAT community. xCDAT supports CDML to enable these users to
+    more easily integrate xCDAT into their existing CDML/CDAT-based workflows.
+
+    References
+    ----------
+    .. [6] https://cdms.readthedocs.io/en/latest/manual/cdms_6.html
+    """
+    # `resolve_entities=False` and `no_network=True` guards against XXE attacks.
+    # Source: https://rules.sonarsource.com/python/RSPEC-2755
+    parser = etree.XMLParser(resolve_entities=False, no_network=True)
+    tree = etree.parse(xml_path, parser)
+    root = tree.getroot()
+
```

Comment:
I'm using `lxml` to with `resolve_entities=False` and `no_network=True` to avoid XXE attacks.
These settings aren't easily configurable with `ElementTree` from the Python standard lib.

More info on XXE attacks:
- https://rules.sonarsource.com/python/RSPEC-2755
- https://codeql.github.com/codeql-query-help/python/py-xxe/
- [Python docs](https://docs.python.org/3/library/xml.etree.elementtree.html) warning that `ElementTree` is not secure